codemp/qcommon/common.cpp

```diff
@@ -289,7 +293,7 @@ void NORETURN QDECL Com_Error( int code, const char *fmt, ... ) {
 	}
 
 	if ( code == ERR_DISCONNECT || code == ERR_SERVERDISCONNECT || code == ERR_DROP || code == ERR_NEED_CD ) {
-		throw code;
+		longjmp(abortframe, code+1); // +1 to avoid 0 value
```

Comment:
I think using longjmp instead of exceptions should probably be a separate PR. I have many thoughts.

Reply:
I did not understand that exceptions cause crashes in QVM JIT. That makes sense, and explains why its necessary to revert to the longjmp used in JK2 and Q3. I hate that this prevents the use of custom destructors in any function that can (directly or indirectly) call `Com_Error`, but translating between longjmps and exceptions at the module boundary every time does not seem feasible.

- [ ] I should annotate all functions that are affected by this to see the impact and check if it's safe. I'd kind of like those annotations to be included in this PR, but I suspect there will be too many of them, leading to many potential merge conflicts downstream.